### DISCUSSION TITLE
Remote Submit in React Hook Form

### DISCUSSION THREAD

**Discussion by NasrinLatifi** [react-hook-form/react-hook-form#8212] [General]
Hi guys!
I'm using React Hook Form. And I need to submit my form from another FC (Function Component).
I read the code of https://codesandbox.io/s/remote-handlesubmit-ipbqb?file=/src/App.js:490-1007, but it is submitted from the component that contains the form. But I need to submit it from another component.
Could you please guide me?

**Comment by bluebill1049** [comment]
see this answer: https://github.com/react-hook-form/react-hook-form/issues/566#issuecomment-978077316 may helps you.

**Reply by NasrinLatifi** [reply]
Thanks a lot, but I want to handle it by js and without a button.

**Comment by Moshyfawn** [comment]
Just pass the `onSubmit` callback from a parent component into your `Form` component. If you need to have the submit button outside the form, you can add an `id` attribute to the form and pass the same form `id` to your submit button to it's `form` attribute.
```jsx
const formId = "form-id"

return (
  <>
    <form id={formId} onSubmit={handleSubmit(onSubmit)}></form>
    <button type="submit" form={formId}>Submit</button>
  </>
)

```

**Reply by NasrinLatifi** [reply]
Thanks a lot, but I want to handle it by js and without a button.

**Reply by bluebill1049** [reply]
`handleSubmit(onSubmit)()`

**Reply by NasrinLatifi** [reply]
It doesn't work, I tested it. 
It works just when register and handleSubmit  import from same useForm.
 const { register, handleSubmit } = useForm();

**Reply by Moshyfawn** [reply]
What do you mean "by js"?

They only other way I can think of is to have a `ref` of your form and remote execute the form submission.

```js
formRef.current.dispatchEvent(
  new Event('submit', { cancelable: true, bubbles: true })
)
``` 

[See this CSB](https://codesandbox.io/s/remote-handlesubmit-ipbqb?file=%2Fsrc%2FApp.js)

**Reply by Moshyfawn** [reply]
>  It works just when register and handleSubmit import from same useForm.

You have to use the form methods from the same form instance. Other-wise, it won't work.

**Reply by NasrinLatifi** [reply]
Thanks a lot :)

**Reply by KarafiziArtur** [reply]
Is there a way to get the ref of the form from the context?

**Reply by Moshyfawn** [reply]
No

**Reply by KarafiziArtur** [reply]
> No

Do you plan to add the feature to submit the form by a specified handler from the context?

**Comment by DaiNguyen1902** [comment]
You can add a ref to the form component and dispatch a submit event to that form
```
// Init react hook form
const {
    register,
    handleSubmit,
    formState: { errors },
} = useForm();

// Init a ref to the form
const formRef = useRef(null);

// Add formRef to you form
return (
  <form ref={formRef} onSubmit={handleSubmit((data) => console.log(data))}>... </form>
)

// put this code line where you want to call remote submit
formRef.current.dispatchEvent(
    new Event('submit', { cancelable: true, bubbles: true })
)
```
